### PR TITLE
pythonPackages.scan-build: init at 2.0.18

### PR DESCRIPTION
**Pull request by malob**
###### Motivation for this change

`scan-build` is a useful tool for C/C++ development that I use frequently, and it would be nice if it were included in `nixpkgs`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4391,6 +4391,12 @@
     githubId = 115218;
     name = "Felix Richter";
   };
+  malo = {
+    email = "mbourgon@gmail.com";
+    github = "malob";
+    githubId = 2914269;
+    name = "Malo Bourgon";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: should be a separate commit:
```
maintainers: add malo
```

File: pkgs/development/python-modules/scan-build/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  pname = "scan-build";
+  version = "2.0.18";
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "296bb899dc4c126b985fca0c15c9ec20cf758b7c7ff830db64c4ca971d886797";
+  };
+
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/scan-build/default.nix]: please checkout from github and run unit tests https://github.com/rizsotto/scan-build/tree/master/tests/unit

File: pkgs/development/python-modules/scan-build/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/rizsotto/scan-build";
+    license = licenses.ncsa;
+    description = "static code analyzer wrapper for Clang.";
```
Comment (jonringer) [inline pkgs/development/python-modules/scan-build/default.nix]:
```suggestion
    description = "Static code analyzer wrapper for Clang";
```

File: pkgs/development/python-modules/scan-build/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/scan-build/default.nix]: important for testing regressions

File: pkgs/development/python-modules/scan-build/default.nix (hunk 3 above)
Comment (malob) [reply pkgs/development/python-modules/scan-build/default.nix]: Done.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (malob) [reply maintainers/maintainer-list.nix]: Done.

File: pkgs/development/python-modules/scan-build/default.nix (hunk 2 above)
Comment (malob) [reply pkgs/development/python-modules/scan-build/default.nix]: Alrighty, so I've done this by changing substituting `doCheck = false;` with:

```
  checkInputs = [ nose ];
  checkPhase = "nosetests tests/unit";
```

However, now I'm getting a test failure, and I'm not sure what I need to do to get it to pass. This is the output of the failed test:

```
..........................F...............................S..S.S.S.S.....S.S
======================================================================
FAIL: test_get_clang_arguments_fails (test_clang.ClangGetArgumentsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-python3.7-scan-build-2.0.18.drv-0/source/tests/unit/test_clang.py", line 38, in test_get_clang_arguments_fails
    sut.get_arguments(['clang', '-x', 'c', 'notexist.c'], '.')
AssertionError: Exception not raised
-------------------- >> begin captured logging << --------------------
root: DEBUG: exec command ['clang', '-###', '-x', 'c', 'notexist.c'] in /private/tmp/nix-build-python3.7-scan-build-2.0.18.drv-0/source
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 76 tests in 1.768s

FAILED (SKIP=7, failures=1)
```

I've checked simple things like check that `clang` is in the build environment. Not sure where to go from here.

File: pkgs/development/python-modules/scan-build/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub{
+    owner = "rizsotto";
+    repo = pname;
+    rev = version;
+    sha256 = "1fpmpg41314qwy6prwknh0hkclcqc1lm7lbmh4r8scsfzs3zch6b";
+  };
+
+  checkInputs = [ nose ];
+  checkPhase = "nosetests tests/unit";
```
Comment (jonringer) [inline pkgs/development/python-modules/scan-build/default.nix]: not sure what the nose equivalent would be, but I would do this in pytest:
```
  # ignore odd clang tests
  checkInputs  =  [ pytest clang ];
  checkPhase = ''
    pytest tests/unit -k 'not sip and not get_clang_arguments_fails'
  '';
```

**Review by jonringer — COMMENTED** [review]
this is what I would do

File: pkgs/development/python-modules/scan-build/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/scan-build/default.nix]: see my other comment.

clang, if listed as a build input, will not be present during checkPhase

File: pkgs/development/python-modules/scan-build/default.nix (hunk 4 above)
Comment (malob) [reply pkgs/development/python-modules/scan-build/default.nix]: Alrighty, I've done the equivalent, tests are now passing.

Comment (jonringer) [reply pkgs/development/python-modules/scan-build/default.nix]: I disabled two tests, the 'sip' test is still failing:
```
builder for '/nix/store/5srpjrm27mfscprqxipgf07f64pd1rsy-python3.8-scan-build-2.0.18.drv' failed with exit code 1; last 10 log lines:
      self.assertTrue(sut.is_preload_disabled(osx))
  AssertionError: False is not true
  -------------------- >> begin captured logging << --------------------
  root: DEBUG: exec command ['csrutil', 'status'] in /build/source
  --------------------- >> end captured logging << ---------------------

  ----------------------------------------------------------------------
  Ran 74 tests in 0.676s

  FAILED (SKIP=7, failures=1)
builder for '/nix/store/1lh6pv3c46knrm0pscqzvabd8yc9pkl3-python3.7-scan-build-2.0.18.drv' failed with exit code 1; last 10 log lines:
      self.assertTrue(sut.is_preload_disabled(osx))
  AssertionError: False is not true
  -------------------- >> begin captured logging << --------------------
  root: DEBUG: exec command ['csrutil', 'status'] in /build/source
  --------------------- >> end captured logging << ---------------------

  ----------------------------------------------------------------------
  Ran 74 tests in 0.696s

  FAILED (SKIP=7, failures=1)
cannot build derivation '/nix/store/i2h96aq1cqbhffdm8pwvmrhcc8v1jmz9-env.drv': 2 dependencies couldn't be built
[1 built (2 failed), 6 copied (843.8 MiB), 138.0 MiB DL]
error: build of '/nix/store/i2h96aq1cqbhffdm8pwvmrhcc8v1jmz9-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/80374
2 package failed to build:
python37Packages.scan-build python38Packages.scan-build
```

Comment (malob) [reply pkgs/development/python-modules/scan-build/default.nix]: Huh, that's not failing for me. I'll look into it.

Comment (jonringer) [reply pkgs/development/python-modules/scan-build/default.nix]: not sure where scrutil is coming from
